test/v1/models/response/liability/LiabilitySpec.scala

```diff
@@ -0,0 +1,164 @@
+              )
+            ),
+            `type` = "VAT",
+            originalAmount = 463872,
+            outstandingAmount = Some(463872),
+            due = Some("2017-03-08")
+          )
+
+        desJson.as[Liability] shouldBe liability
+      }
+
+      "one liability is returned without items:dueDate" in {
+
+        val desJson = Json.parse(
+          s"""
+             |{
+             |            "chargeType": "VAT",
+             |            "mainType": "2100",
+             |            "periodKey": "13RL",
+             |            "periodKeyDescription": "abcde",
+             |            "taxPeriodFrom": "2017-01-01",
+             |            "taxPeriodTo": "2017-04-05",
+             |            "businessPartner": "6622334455",
+             |            "contractAccountCategory": "02",
+             |            "contractAccount": "D",
+             |            "contractObjectType": "ABCD",
+             |            "contractObject": "00000003000000002757",
+             |            "sapDocumentNumber": "1040000872",
+             |            "sapDocumentNumberItem": "XM00",
+             |            "chargeReference": "XM002610011594",
+             |            "mainTransaction": "1234",
+             |            "subTransaction": "5678",
+             |            "originalAmount": 463872,
+             |            "outstandingAmount": 463872,
+             |            "accruedInterest": 10000
+             |}
+             |""".stripMargin)
+
+        val liability = Liability(
+            taxPeriod = Some(
+              TaxPeriod(
+                from = "2017-01-01",
+                to = "2017-04-05"
+              )
+            ),
+            `type` = "VAT",
+            originalAmount = 463872,
+            outstandingAmount = Some(463872),
+            due = None
+          )
+
+        desJson.as[Liability] shouldBe liability
+      }
+
+      "not parse incorrect json" in {
+
+        val desJson = Json.parse(
+          s"""
+             |{
+             |    "idType": "VRN"
+             |}
+             |""".stripMargin)
+
+        desJson.asOpt[Liability] shouldBe None
+      }
+
+    }
+
+    "use the writes format correctly" in {
+
+      val desJson = Json.parse(
+        s"""
+           |{
+           |		"taxPeriod": {
+           |			"from": "2017-01-01",
+           |			"to": "2017-04-05"
+           |		},
+           |		"type": "VAT",
+           |		"originalAmount": 463872,
+           |		"outstandingAmount": 463872,
+           |		"due": "2017-03-08"
+           |	}
+           |""".stripMargin)
+
+      val liability = Liability(
+        taxPeriod = Some(
+          TaxPeriod(
+            from = "2017-01-01",
+            to = "2017-04-05"
+          )
+        ),
+        `type` = "VAT",
+        originalAmount = 463872,
+        outstandingAmount = Some(463872),
+        due = Some("2017-03-08")
+      )
+
+      Json.toJson(liability) shouldBe desJson
```

Comment:
does Json **writes** converts to desJson??

Reply:
It's just a mis-named val, will rename now